pkg/kubelet/status/generate_test.go

```diff
@@ -115,6 +115,103 @@ func TestGeneratePodReadyCondition(t *testing.T) {
 	}
 }
 
+func TestGeneratePodInitializedCondition(t *testing.T) {
+	tests := []struct {
+		spec              *v1.PodSpec
+		containerStatuses []v1.ContainerStatus
+		podPhase          v1.PodPhase
+		expected          v1.PodCondition
+	}{
+		{
+			spec: &v1.PodSpec{
+				InitContainers: []v1.Container{
+					{Name: "1234"},
+					{Name: "5678"},
+				},
+			},
+			containerStatuses: nil,
+			podPhase:          v1.PodRunning,
+			expected:          getInitializedCondition(false, "UnknownContainerStatuses", ""),
+		},
+		{
+			spec:              &v1.PodSpec{},
+			containerStatuses: []v1.ContainerStatus{},
+			podPhase:          v1.PodRunning,
+			expected:          getInitializedCondition(true, "", ""),
+		},
+		{
+			spec: &v1.PodSpec{
+				InitContainers: []v1.Container{
+					{Name: "1234"},
+				},
+			},
+			containerStatuses: []v1.ContainerStatus{},
+			podPhase:          v1.PodRunning,
+			expected:          getInitializedCondition(false, "ContainersNotInitialized", "containers with unknown status: [1234]"),
+		},
+		{
+			spec: &v1.PodSpec{
+				InitContainers: []v1.Container{
+					{Name: "1234"},
+					{Name: "5678"},
+				},
+			},
+			containerStatuses: []v1.ContainerStatus{
+				getReadyStatus("1234"),
+				getReadyStatus("5678"),
+			},
+			podPhase: v1.PodRunning,
+			expected: getInitializedCondition(true, "", ""),
+		},
+		{
+			spec: &v1.PodSpec{
+				InitContainers: []v1.Container{
+					{Name: "1234"},
+					{Name: "5678"},
+				},
+			},
+			containerStatuses: []v1.ContainerStatus{
+				getReadyStatus("1234"),
+			},
+			podPhase: v1.PodRunning,
+			expected: getInitializedCondition(false, "ContainersNotInitialized", "containers with unknown status: [5678]"),
+		},
+		{
+			spec: &v1.PodSpec{
+				InitContainers: []v1.Container{
+					{Name: "1234"},
+					{Name: "5678"},
+				},
+			},
+			containerStatuses: []v1.ContainerStatus{
+				getReadyStatus("1234"),
+				getNotReadyStatus("5678"),
+			},
+			podPhase: v1.PodRunning,
+			expected: getInitializedCondition(false, "ContainersNotInitialized", "containers with incomplete status: [5678]"),
+		},
+		{
+			spec: &v1.PodSpec{
+				InitContainers: []v1.Container{
+					{Name: "1234"},
+				},
+			},
+			containerStatuses: []v1.ContainerStatus{
+				getNotReadyStatus("1234"),
```

Comment:
this case covers 
https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/status/generate.go#L105-L111

 could you tell me how to fix ?